SU2_CFD/include/fluid/CFluidScalar.hpp

```diff
@@ -43,13 +43,58 @@ class CFluidScalar final : public CFluidModel {
   su2double Gamma;                  /*!< \brief Ratio of specific heats of the gas. */
   su2double Pressure_Thermodynamic; /*!< \brief Constant pressure thermodynamic. */
 
-  std::vector<su2double> molarMasses;  /*!< \brief Molar masses of all species. */
+  bool wilke;
+  bool davidson;
+
+  std::vector<su2double> massFractions;              /*!< \brief Mass fractions of all species. */
+  std::vector<su2double> moleFractions;              /*!< \brief Mole fractions of all species. */
+  std::vector<su2double> molarMasses;                /*!< \brief Molar masses of all species. */
+  std::vector<su2double> laminarViscosity;           /*!< \brief Laminar viscosity of all species. */
+  std::vector<su2double> laminarThermalConductivity; /*!< \brief Laminar thermal conductivity of all species. */
+
+  static const int ARRAYSIZE = 100;
```

Comment:
```suggestion
  static constexpr int ARRAYSIZE = 16;
```
Use something reasonable here, if you tried to use 100 species the code would run 10000 times slower